test/functional/tests/fault_injection/test_error_device_as_cache.py

```diff
@@ -0,0 +1,280 @@
+def test_error_device_as_cache_clean(cache_mode, cache_line_size):
+    """
+    title: Validate Open CAS ability to handle read hit I/O error on cache device for clean data
+    description: |
+        Perform I/O on exported object while error device is cache device and validate if
+        errors are present in Open CAS stats.
+    pass_criteria:
+      - Write error count in io is zero
+      - Read error count in io is zero
+      - Write error count in cache statistics is zero
+      - Total error count in cache statistics is greater than zero
+    """
+
+    with TestRun.step("Prepare cache and core devices"):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(500, Unit.MebiByte)])
+        core_device.create_partitions([Size(400, Unit.MebiByte)])
+
+        cache_part = cache_device.partitions[0]
+        error_device = ErrorDevice("error", cache_part)
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(
+            error_device, cache_mode=cache_mode, cache_line_size=cache_line_size, force=True
+        )
+
+    with TestRun.step("Disable cleaning policy and sequential cutoff"):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step(f"Add core"):
+        core = cache.add_core(core_dev=core_device.partitions[0])
+
+    if cache_mode == CacheMode.WA:
+        with TestRun.step("Read core to null block"):
+            dd = (
+                Dd()
+                .input(core)
+                .output("/dev/null")
+                .iflag("direct")
+                .block_size(cache_line_size.value)
+            )
+            dd.run()
+
+    with TestRun.step("Run fio against core to fill it with pattern"):
+        fio = (
+            Fio()
+            .create_command()
+            .target(core)
+            .io_engine(IoEngine.libaio)
+            .io_depth(1)
+            .num_jobs(1)
+            .size(cache_part.size)
+            .read_write(ReadWrite.randwrite)
+            .block_size(cache_line_size)
+            .rand_seed(int(cache_part.size.get_value()))
+            .direct()
+            .verification_with_pattern("0xabcd")
+            .do_verify(False)
+            .continue_on_error(ErrorFilter.io)
+        )
+        fio_errors = fio.run()[0].total_errors()
+
+    with TestRun.step("Check if fio reported no write errors"):
+        if fio_errors != 0:
+            TestRun.fail("Fio reported write errors!")
+
+    with TestRun.step("Stop cache"):
+        metadata_size = cache.get_metadata_size_on_disk() + Size(1, Unit.MiB)
+        cache.stop()
+
+    with TestRun.step("Enable errors on cache device(after metadata area)"):
+        error_device.change_table(
+            error_table(offset=cache_part.size, size=metadata_size).fill_gaps(cache_part)
```

Comment:
Size and offset should be probably swapped. And please change prameter names to start and stop lba